src/apify_client/clients/base/base_client.py

```diff
@@ -1,8 +1,13 @@
-from typing import Any, Dict, Optional
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Dict, Optional
 
 from ..._http_client import _HTTPClient
 from ..._utils import _to_safe_id
 
+if TYPE_CHECKING:
+    from ...client import ApifyClient
```

Comment:
Why this import pls? Conditional imports are scary, would it be able to avoid it?

Reply:
It's a conditional import only executed when type checking, otherwise we'd get circular dependency issues during runtime. It's a pretty standard way of doing things recommended by the core typing module, I don't think it's that scary.